nano/node/json_handler.cpp

```diff
@@ -4452,7 +4452,8 @@ void nano::json_handler::work_generate ()
 		}
 		else
 		{
-			node.work_generate (hash, callback, difficulty);
+			// Double XOR with nano::node::work_generate () function
+			node.work_generate (node.network_params.network.is_live_network () ? hash : hash ^ static_cast<uint64_t> (node.network_params.network.current_network), callback, difficulty);
```

Comment:
It seems like what's applied here would be undone inside work_generate.

Should the xoring be done in the wallet when work is queued?

Reply:
Wallet also uses work_generate () & many tests contain work_generate_blocking etc.
So I guess easier to XOR in single function (work_generate)

But because "work_generate" RPC should receive already XORed value, XOR will be undone by work_generate. With extra XOR here we can pass root to work generator unchanged

Reply:
Can we add an overload which accepts the 64-bit value to mix, which then calls the other work_generate overload? Then the RPC would just directly call the overload without the 64bit value.

Reply:
Done